go/vt/vttablet/tabletserver/throttle/throttler.go

```diff
@@ -1059,22 +926,40 @@ func (throttler *Throttler) generateTabletProbeFunction(scope base.Scope, tmClie
 	}
 }
 
+// readSelfThrottleMetricsInternal rreads all registsred self metrics on this tablet (or backend MySQL server).
+// This is the actual place where metrics are read, to be later aggregated and/or propagated to other tablets.
 func (throttler *Throttler) readSelfThrottleMetricsInternal(ctx context.Context) base.ThrottleMetrics {
-
-	writeMetric := func(metricName base.MetricName, metric *base.ThrottleMetric) {
-		metric.Name = metricName
+	result := make(base.ThrottleMetrics)
```

Comment:
We could allocate the space/capacity using `base.RegisteredSelfMetrics` here, no? I would think that this gets called fairly often in total to we should take advantage of easy optimizations, I think.

Reply:
Fixed.